client-http/src/main/java/com/cloudera/livy/client/http/LivyConnection.java

```diff
@@ -85,6 +114,7 @@ public int getSocketTimeout() {
       .setConnectionReuseStrategy(new DefaultConnectionReuseStrategy())
       .setDefaultRequestConfig(reqConfig)
       .setMaxConnTotal(1)
+      .setDefaultCredentialsProvider(credsProvider)
```

Comment:
No, this is for kerberos. It's described in the httpclient guide for SPNEGO (which basically just tells you to try to make the example they have in their repo work).